contracts/BIFI/strategies/Stargate/StrategyStargateStakingTridentRouter.sol

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+import "../../interfaces/common/ITridentRouter.sol";
+import "../../interfaces/common/IMasterChef.sol";
+import "../../interfaces/stargate/IStargateRouter.sol";
+import "../Common/StratManager.sol";
+import "../Common/FeeManager.sol";
+import "../../utils/StringUtils.sol";
+import "../../utils/GasThrottler.sol";
+
+contract StrategyStargateStakingTridentRouter is StratManager, FeeManager, GasThrottler {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    // Tokens used
+    address public native;
+    address public output;
+    address public want;
+    address public lpToken0;
+
+    // Third party contracts
+    address public chef;
+    uint256 public poolId;
+    address public stargateRouter;
+    uint256 public routerPoolId;
+
+    bool public harvestOnDeposit;
+    uint256 public lastHarvest;
+    string public pendingRewardsFunctionName;
+
+    // Routes
+    address[][] public outputToNativeRoute;
+    address[][] public outputToLp0Route;
+    address[] public outputToNativePoolRoute;
+    address[] public outputToLp0PoolRoute;
+
+    // paths
+    ITridentRouter.Path[] public outputToNativePath;
+    ITridentRouter.Path[] public outputToLp0Path;
+
+    event StratHarvest(address indexed harvester, uint256 wantHarvested, uint256 tvl);
+    event Deposit(uint256 tvl);
+    event Withdraw(uint256 tvl);
+    event ChargedFees(uint256 callFees, uint256 beefyFees, uint256 strategistFees);
+
+    constructor(
+        address _want,
+        uint256[] memory _poolIdAndRouterPoolId,
```

Comment:
Guessing you ran into the local variable limit here?

Reply:
correct! is this way of packing inputs into arrays appropriate? or would you prefer it different?